webapp/client/src/components/FormFieldYesNo.js

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+import FieldLabelForSeparatedFields from "./FieldLabelForSeparatedFields";
+
+const YesNo = styled.div`
+  & .switch-yes,
+  .switch-no {
+    padding: 1rem 1.45rem;
+
+    color: var(--text-color) !important;
+    background: var(--bg-white) !important;
+    font-size: var(--text-s);
+    box-shadow: none;
+    border-radius: 0;
+    border: 1px solid var(--border-color) !important;
+  }
+
+  & .switch-no {
+    margin-left: -1px; // To hide double border in the middle
+  }
+
+  & .switch-yes:focus-within:not(:hover),
+  .switch-no:focus-within:not(:hover) {
+    padding: 1rem 1.45rem calc(1rem - 3px) 1.45rem;
+    position: relative;
+    z-index: 1;
+
+    color: var(--focus-text-color) !important;
+    background: var(--focus-color) !important;
+
+    border: 1px solid var(--border-color) !important;
+    border-bottom: 0px solid black !important;
+    box-shadow: 0px 4px 0px var(--text-color);
+  }
+
+  & .switch-yes.active,
+  .switch-no.active {
+    color: var(--inverse-text-color) !important;
+    background: var(--link-color) !important;
+    border-radius: 0;
+    border: 1px solid var(--link-color) !important;
+  }
+
+  & .switch-yes:active,
+  .switch-no:active {
+    box-shadow: 0 0 0 3px var(--focus-color) !important;
+  }
+
+  & .switch-yes:hover,
+  .switch-no:hover,
+  .switch-yes.active:hover,
+  .switch-no.active:hover,
+  .switch-yes:focus-within:hover,
+  .switch-no:focus-within:hover {
+    color: var(--inverse-text-color) !important;
+
+    background: var(--link-hover-color) !important;
+    border: 1px solid var(--link-hover-color) !important;
+  }
+
+  & input[type="radio"] {
+    opacity: 0;
+    position: absolute;
+  }
+`;
+
+function FormFieldYesNo({
+  fieldName,
+  fieldId,
+  value,
+  required,
+  autofocus,
+  label,
+  details,
+  errors,
+  onChangeHandler,
+}) {
+  const [selectedValue, setSelectedValue] = useState(value);
+  const yesFieldId = `${fieldId}-yes`;
+  const noFieldId = `${fieldId}-no`;
+
+  const toggleYesNoButton = (event) => {
+    setSelectedValue(event.target.value);
+  };
+
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+        cols: "6",
+      }}
+      hideLabel
+      render={() => (
+        <YesNo>
+          <FieldLabelForSeparatedFields {...{ label, fieldId, details }} />
+          <fieldset id={fieldId} name={fieldId}>
+            <label
+              htmlFor={yesFieldId}
+              className={`switch-yes ${
```

Comment:
https://www.npmjs.com/package/classnames makes conditionally setting classes much easier. See here for an example: https://github.com/digitalservice4germany/steuerlotse/blob/650042b632a6fffb9979f2e1b53d36456e779604/webapp/client/src/components/FormFieldScaffolding.js#L49

Reply:
Thanks that is a great idea and is so much better.